packages/kit/src/runtime/server/respond.js

```diff
@@ -508,11 +508,9 @@ export async function respond(request, options, manifest, state) {
 			}
 
 			if (state.error && event.isSubRequest) {
-				return await fetch(request, {
-					headers: {
-						'x-sveltekit-error': 'true'
-					}
-				});
+				const headers = new Headers(request.headers);
```

Comment:
```suggestion
        // avoid overwriting the headers. This could be a same origin fetch request
        // to an external service from the root layout while rendering an error page
				const headers = new Headers(request.headers);
```